app/components/provider_interface/safeguarding_declaration_component.rb

```diff
@@ -2,13 +2,41 @@ module ProviderInterface
   class SafeguardingDeclarationComponent < ViewComponent::Base
     include ViewHelper
 
-    attr_reader :message
+    attr_reader :application_choice, :current_provider_user
 
-    def initialize(application_form:)
-      @message = SafeguardingStatus.new(
-        status: application_form.safeguarding_issues_status,
+    def initialize(application_choice:, current_provider_user:)
+      @application_choice = application_choice
+      @current_provider_user = current_provider_user
+    end
+
+    def message
+      SafeguardingStatus.new(
+        status: status,
         i18n_key: 'provider_interface.safeguarding_declaration_component',
       ).message
     end
+
+    def display_safeguarding_information?
+      current_provider_user.authorisation
+        .can_view_safeguarding_information?(course: application_choice.course)
+    end
+
+    def details
+      application_choice.application_form.safeguarding_issues
+    end
+
+  private
+
+    def status
+      if safeguarding_issues_declared? && !display_safeguarding_information?
+        'has_safeguarding_issues_to_declare_no_permissions'
+      else
```

Comment:
Double checking: is it correct that if the user hasn't got permission but issues are not declared that we still show the status? i.e. if you don't have permission you are still told that there are no issues.

Reply:
That's my understanding based on [the design history entry](https://bat-design-history.netlify.app/manage-teacher-training-applications/view-safeguarding-information/).

Reply:
ok great :+1: